cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -321,6 +400,38 @@ private String getPluginName(Class<?> cls) {
   }
 
   /**
+   * Extracts and returns set of endpoints in the plugin.
+   * @throws IllegalArgumentException if there are duplicate endpoints found (or)
+   * if the number of arguments is not 1 or 2, or if type of 2nd argument is not an EndpointPluginContext.
+   */
+  private Set<String> getPluginEndpoints(Class<?> cls) throws IllegalArgumentException {
+    Set<String> endpoints = new HashSet<>();
+    Method[] methods = cls.getMethods();
+    for (Method method : methods) {
+      Path pathAnnotation = method.getAnnotation(Path.class);
+      // method should have path annotation else continue
+      if (pathAnnotation != null) {
+        if (!endpoints.add(pathAnnotation.value())) {
+          // if the endpoint already exists throw an exception saying, plugin has two methods with same endpoint name.
+          throw new IllegalArgumentException(String.format("Two Endpoints with same name : {} found in Plugin : {}",
+                                                           pathAnnotation.value(), getPluginName(cls)));
+        }
+        // check that length of method parameters is 1 or 2. if length is 2,
+        // check that 2nd param is of type EndpointPluginContext
+        if (!(method.getParameterTypes().length == 1 || method.getParameterTypes().length == 2)) {
+          throw new IllegalArgumentException(String.format("Endpoint parameters can only be of length 1 or 2"));
```

Comment:
this would be a confusing error message if I saw it.  Would help to include the class and method